src/ray/gcs/gcs_server/gcs_job_manager.cc

```diff
@@ -147,21 +166,29 @@ void GcsJobManager::HandleMarkJobFinished(rpc::MarkJobFinishedRequest request,
                                           rpc::SendReplyCallback send_reply_callback) {
   const JobID job_id = JobID::FromBinary(request.job_id());
 
-  auto send_reply = [send_reply_callback, reply](Status status) {
+  auto send_reply = [send_reply_callback = std::move(send_reply_callback),
+                     reply](Status status) {
     GCS_RPC_SEND_REPLY(send_reply_callback, reply, status);
   };
 
   Status status = gcs_table_storage_->JobTable().Get(
       job_id,
-      [this, job_id, send_reply](Status status,
+      [this, job_id, send_reply](const Status &status,
                                  const std::optional<rpc::JobTableData> &result) {
         if (status.ok() && result) {
           MarkJobAsFinished(*result, send_reply);
-        } else {
+          ++finished_jobs_count_;
```

Comment:
I mean we need to move this into the body of MarkJobAsFinished. Specifically, put it besides `running_job_ids_.erase(iter);`. The reason is MarkJobAsFinished is not only called here; it's also called in `GcsJobManager::OnNodeDead` where we also want to update the count.

Reply:
Updated.